File: Objective-C Version/JumpStarDemo/CoreAnimations/ViewController.m
//
//  ViewController.m
//  CoreAnimations
//
//  Created by Kitten Yang on 9/16/15.
//  Copyright (c) 2015 Kitten Yang. All rights reserved.
//

#import "ViewController.h"
#import "JumpStarView.h"
@interface ViewController ()
@property (weak, nonatomic) IBOutlet JumpStarView *jumpStarView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [_jumpStarView layoutIfNeeded];
    _jumpStarView.markedImage = [UIImage imageNamed:@"icon_star_incell"];
    _jumpStarView.non_markedImage = [UIImage imageNamed:@"blue_dot"];
    _jumpStarView.state = NONMark;
}

- (IBAction)tapped:(id)sender {
    
    [_jumpStarView animate];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift Version/JumpStarDemo-Swift/JumpStarDemo-Swift/ViewController.swift
//
//  ViewController.swift
//  JumpStarDemo-Swift
//
//  Created by Kitten Yang on 1/6/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var jumpStarView: JumpStarView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        jumpStarView.layoutIfNeeded()
        let option = JumpStarOptions(markedImage: UIImage(named: "icon_star_incell")!, notMarkedImage: UIImage(named: "blue_dot")!, jumpDuration: 0.125, downDuration: 0.215)
        jumpStarView.option = option
        jumpStarView.state = .NotMarked
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func jumpButtonDidTapped(sender: AnyObject) {
        jumpStarView.animate()
    }

}

